### PR TITLE
Project-Mongo-API - Erika Andersson Porath

### PR DESCRIPTION
**Pull request by TheErry**
This project was once again very straight forward and I haven't faced many problems with it. I went for the same data as in the last project so I already knew what endpoints I could use. I decided to take the weekend off so I haven't approached more than the blue goals. One error message is missing because I couldn't get it to work though.

### CODE REVIEW

File: server.js [hunk 1]
```diff
@@ -1,38 +1,107 @@
...
+})
+
+// List all shows (all data)
+app.get("/shows", async (_, res) => {
+  // async/await version
+  const shows = await Show.find()
+  res.json(shows)
+})
+
+// Look for a specific show by using Mongo-ID
```
Comment (AsaHildebrand) [inline server.js:72]: Nice with the explanations of how the endpoints works :)

File: README.md [hunk 2]
```diff
@@ -1,13 +1,23 @@
...
+
+**The aviable endpoints are:**
+
+/ - *lists all endpoints*
+
+/shows - *lists all the shows*
+
+/shows/:id - *endpoint to look for a specific shoe by using the Mongo-ID*
+
+/shows/title/:showTitle - *endpoint to look for a specific show by using the title*
```
Comment (AsaHildebrand) [inline README.md:19]: Nice with this list so you can see which endpoints that are used in the API 👍

File: server.js [hunk 3]
```diff
@@ -1,38 +1,107 @@
...
-// Add middlewares to enable cors and json body parsing
+// Add middlewares to enable cors, json body parsing and a error status if the database is down
 app.use(cors())
-app.use(bodyParser.json())
+app.use(express.json())
+app.use((_, res, next) => {
+  if (mongoose.connection.readyState === 1) {
+    next()
+  } else {
+    res.status(503).json({ error: "Service unavailable" })
```
Comment (AsaHildebrand) [inline server.js:54]: Great with middleware for server error :)

**Review by AsaHildebrand — APPROVED** [review]
Nice API! 😃 You have got a clean code, easy to read, with good explanations. Good job with the endpoints and the error messages 🥇

File: README.md (hunk 2 above)
Comment (KarolCislowski) [inline README.md:19]: Nice short documentation 👍🏻

File: server.js [hunk 4]
```diff
@@ -1,38 +1,107 @@
...
+
+import netflixData from "./data/netflix-titles.json"
 
 const mongoUrl = process.env.MONGO_URL || "mongodb://localhost/project-mongo"
 mongoose.connect(mongoUrl, { useNewUrlParser: true, useUnifiedTopology: true })
 mongoose.Promise = Promise
 
+// Mongoose model. No showSchema because it's only going to be used here 
+const Show = mongoose.model("Show", {
+  show_id: Number,
```
Comment (KarolCislowski) [inline server.js:14]: Why do you need both show_id and _id?

**Review by KarolCislowski — APPROVED** [review]
Nice API 👍🏻

**Pull request closed without merging**